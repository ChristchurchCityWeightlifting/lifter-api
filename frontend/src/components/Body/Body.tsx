/** @format */

import React from "react";
import { Typography } from "@mui/material";

interface BodyProps {
  children?: React.ReactNode;
}

const Body: React.FC<React.PropsWithChildren<BodyProps>> = (
  props: BodyProps
) => {
  return (
    <Typography variant="body1" gutterBottom noWrap {...props}>
      {props.children}
    </Typography>
  );
};

export default Body;
