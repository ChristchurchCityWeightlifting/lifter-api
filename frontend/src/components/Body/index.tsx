/** @format */

export { default } from "./Body";
