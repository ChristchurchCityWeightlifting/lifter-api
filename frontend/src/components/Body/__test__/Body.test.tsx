/** @format */

import React from "react";
import { describe, test, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import Body from "../Body";

describe("Title component tests", () => {
  test("renders without text", () => {
    const tree = renderer.create(<Body></Body>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders with text", () => {
    const tree = renderer.create(<Body>Test Text</Body>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders with text and props", () => {
    const tree = renderer
      .create(<Body sx={{ textAlign: "left" }}>Test Text</Body>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
