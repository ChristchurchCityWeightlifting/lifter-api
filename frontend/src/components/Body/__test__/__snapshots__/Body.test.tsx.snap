// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component tests renders with text 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1tqvnak-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;margin-bottom:0.35em;}",
      }
    }
    data-emotion="css 1tqvnak-MuiTypography-root"
  />,
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-noWrap css-1tqvnak-MuiTypography-root"
  >
    Test Text
  </p>,
]
`;

exports[`Title component tests renders with text and props 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-o2svw5-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;margin-bottom:0.35em;text-align:left;}",
      }
    }
    data-emotion="css o2svw5-MuiTypography-root"
  />,
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-noWrap css-o2svw5-MuiTypography-root"
  >
    Test Text
  </p>,
]
`;

exports[`Title component tests renders without text 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1tqvnak-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;margin-bottom:0.35em;}",
      }
    }
    data-emotion="css 1tqvnak-MuiTypography-root"
  />,
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiTypography-noWrap css-1tqvnak-MuiTypography-root"
  />,
]
`;
